_pages/dev/event-service.md

```diff
@@ -124,3 +125,339 @@ onClick() {
   this.events.dispatch(new CustomUIEvent(...));
 }
 ```
+
+## Event Types Inheritance
+
+In Spartacus 3.1, we've introduced a concept of "umbrella" events. The purpose of these events is to group similar events under one common umbrella event. This enables you to observe only the umbrella event, and receive all the grouped events.
+
+For example, `@spartacus/core`'s `PageEvent` is an umbrella event. All the specific page events inherit from it (`HomePageEvent`, `CartPageEvent`, etc.).
+In the previous Spartacus versions, you had to observe each of the page events individually (i.e. `eventService.get(HomePageEvent).subscribe(...)`, `eventService.get(CartPageEvent).subscribe(...)`, etc.).
+Starting from Spartacus 3.1, you can subscribe to all the grouped events just by observing the umbrella event:
+
+```typescript
+eventService.get(PageEvent).subscribe(...) // receives all page events
+```
+
+Note that all events _should_ inherit from `@spartacus/core`'s `CxEvent`:
+
+```typescript
+import { CxEvent } from "@spartacus/core";
+export class MyEvent extends CxEvent {...}
+```
+
+This is not required for the event to work properly, but it is desireable to have in cases you want to observe all events at once:
+
+```typescript
+eventService.get(CxEvent).subscribe(...) // receives all Spartacus' events.
+```
+
+_NOTE: observing all Spartacus' events at once might have a significant performance hit, therefore use with care._
+
+## Spartacus Event List
+
+This is the complete list of all the Spartacus events:
```

Comment:
I don't think we should include the complete list of Spartacus events in the written documentation.. We have generated documentation, where it's easy to find any event, check it's shape, etc.
It looks redundant, cumbersome to update, and easy to become obsolete IMO, And maybe even less readable than in generated docs.

Reply:
Discussed offline

Reply:
My first thought on seeing this complete list of events in the docs is to remove it and point to the actual source of truth, for all of the reasons stated in the original comment above.

So this comment is just to confirm that the list of events will be replaced by a link to the generated documentation 🙂